crates/sui-framework/packages/sui-framework/sources/url.move

```diff
@@ -1,10 +1,12 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+#[allow(deprecated_usage)]
```

Comment:
A bit ridiculous, but every function in the `sui::url` uses the deprecated `Url` type. 😆

Reply:
@tzakian, I wonder if we should just suppress deprecation usage within the same package... Or maybe somehow with public functions? Not sure. 
My point is that these warnings here are not-actionable. We cannot break the API

Reply:
Hmmm I'm of two minds on it.

One one hand warning about deprecations on APIs that you cannot change is quite annoying. On the other hand, this might be the only way folks find out about the deprecation, and that they may want to add new functions that use the new type (etc). 

I kinda would lean towards the second opinion the most here (we should warn, and folks need to manually turn it off).